src/backend/commands/createas.c

```diff
@@ -325,13 +327,13 @@ ExecCreateTableAs(ParseState *pstate, CreateTableAsStmt *stmt,
 		check_ivm_restriction_context ctx = {false, false, false, false, NIL, NIL};
 
 		/* check if the query is supported in IMMV definition */
-		if(contain_mutable_functions((Node *)query))
+		if(contain_mutable_functions((Node *) query))
 			ereport(ERROR,
 					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 					 errmsg("mutable function is not supported on incrementally maintainable materialized view"),
 					 errhint("functions must be marked IMMUTABLE")));
 
-		check_ivm_restriction_walker((Node *) query, &ctx, 0);
+		check_ivm_restriction((Node *) query, &ctx);
 
```

Comment:
How about to make  check_ivm_restriction_context ctx in check_ivm_restriction() and remove the ctx argument from this function?

Reply:
That's a nice idea ! fix it.